summingbird-online/src/main/scala/com/twitter/summingbird/online/option/AllOpts.scala

```diff
@@ -98,8 +99,22 @@ trait SummerBuilder extends Serializable {
 
 /**
  * The SummerConstructor option, set this instead of CacheSize, AsyncPoolSize, etc.. to provide how to construct the aggregation for this bolt
+ * @see [[Summers]] for useful [[SummerConstructor]]s.
  */
-case class SummerConstructor(get: SummerBuilder)
+case class SummerConstructor(get: SummerConstructorSpec)
+
+trait SummerConstructorSpec {
+  def builder(counter: Name => Counter with Incrementor): SummerBuilder
```

Comment:
tbh, I think we should just change the definition of SummerBuilder to take the counterBuilder as argument and default to a no-op counterBuilder. SummerBuilder is used very rarely and I think it's ok to break it. There are only a couple places that we'll need to fix at Twitter. Seems better than adding a level of indirection here.
@johnynek wdyt?